# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSet.h
# Commit: fbfb5b7dfb30
# Full Hash: fbfb5b7dfb30b42d6debd6f2808b0f8e8bb4f868
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 2. Merge FontFaceSetImpl and gfxUserFontSet into a single class. r=emilio
#   
#   This patch merges FontFaceSetImpl and gfxUserFontSet into the same
#   class. This reduces the level of indirection and in theory makes it
#   easier to manage future threading issues.
# ==============================================================================

diff -r 93b7e171c74a -r fbfb5b7dfb30 layout/style/FontFaceSet.h
--- a/layout/style/FontFaceSet.h	Tue Jun 28 14:27:20 2022 +0000
+++ b/layout/style/FontFaceSet.h	Tue Jun 28 14:27:21 2022 +0000
@@ -11,13 +11,9 @@
 #include "mozilla/dom/FontFaceSetBinding.h"
 #include "mozilla/dom/FontFaceSetImpl.h"
 #include "mozilla/DOMEventTargetHelper.h"
-#include "gfxUserFontSet.h"
 #include "nsICSSLoaderObserver.h"
 #include "nsIDOMEventListener.h"
 
-struct gfxFontFaceSrc;
-class gfxFontSrcPrincipal;
-class gfxUserFontEntry;
 class nsFontFaceLoader;
 class nsIPrincipal;
 class nsPIDOMWindowInner;
@@ -37,8 +33,6 @@
   friend class mozilla::PostTraversalTask;
 
  public:
-  using UserFontSet = FontFaceSetImpl::UserFontSet;
-
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(FontFaceSet, DOMEventTargetHelper)
 
@@ -67,8 +61,6 @@
 
   void CopyNonRuleFacesTo(FontFaceSet* aFontFaceSet) const;
 
-  UserFontSet* GetUserFontSet() const { return mImpl->GetUserFontSet(); }
-
   void CacheFontLoadability() { mImpl->CacheFontLoadability(); }
 
   FontFaceSetImpl* GetImpl() const { return mImpl; }